### PR TITLE
Progressbar component

### PR DESCRIPTION
**Pull request by k2glyph**
Apart from missing SVG images. Rest has been done. Please do review the code.

### CODE REVIEW

File: src/components/Progressbar.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { FC, PropsWithChildren } from 'react';
+
+export type ProgressbarProps = PropsWithChildren<{
+  progress?: string | '10%' | '20%' | '30%' | '40%' | '50%' | '60%' | '70%' | '80%' | '90%' | '100%';
```
Comment (rluders) [inline src/components/Progressbar.tsx:4]: @k2glyph I know that it still WIP but, guess that would be better to just use number instead string?

Comment (k2glyph) [reply src/components/Progressbar.tsx:4]: okay thanks @rluders. Sure

File: src/components/Progressbar.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import classNames from 'classnames';
+import { FC, PropsWithChildren } from 'react';
+
+type Color = 'gray' | 'blue' | 'red' | 'green' | 'yellow' | 'indigo' | 'purple';
+type Size = 'sm' | 'md' | 'lg' | 'xl';
+
+export type ProgressbarProps = PropsWithChildren<{
+  className?: string;
+  progress?: number;
```
Comment (bacali95) [inline src/components/Progressbar.tsx:9]: I think the progress prop should be required.

File: src/Root.tsx [hunk 3]
```diff
@@ -155,6 +158,12 @@ export const Root: FC = () => {
         title: 'Tooltips',
         href: '/tooltips',
       },
+      {
+        group: false,
+        icon: AiOutlineMinus,
+        title: 'Progressbar',
+        href: '/progressbar',
```
Comment (bacali95) [inline src/Root.tsx:165]: This can be either `progress` or `progress-bar`

File: src/components/Progressbar.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            )}
+          >
+            {label}
+          </span>
+          {labelProgress && (
+            <span
+              className={classNames(
+                'font-medium dark:text-white',
+                labelSizeClasses[labelSize],
+                lableColorClasses[labelColor],
```
Comment (bacali95) [inline src/components/Progressbar.tsx:78]: we can move all calls of `lableColorClasses` and `lableSizeClasses` to the parent suggested `div` to reduce the redundancy in the code.

File: src/pages/ProgressbarPage.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+          <div className="mb-1 text-base font-medium text-purple-700">Purple</div>
+          <Progressbar color="purple" />
+        </>
+      ),
+    },
+    {
+      title: 'With label inside',
+      code: (
+        <>
```
Comment (bacali95) [inline src/pages/ProgressbarPage.tsx:59]: If it is one child we don't need the fragment parent.

File: src/pages/ProgressbarPage.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+          <Progressbar progress={50} label="flowbite" size="lg" />
+        </>
+      ),
+    },
+    {
+      title: 'With label outside',
+      code: (
+        <>
```
Comment (bacali95) [inline src/pages/ProgressbarPage.tsx:67]: Same here

**Review by bacali95 — CHANGES_REQUESTED** [review]
Hello @k2glyph, thanks for your contribution :tulip: 
I just left some review comments, it would be nice if you can check them :smile:

File: src/Root.tsx [hunk 7]
```diff
@@ -159,7 +160,7 @@ export const Root: FC = () => {
       },
       {
         group: false,
-        icon: HiOutlineX,
+        icon: AiOutlineMinus,
```
Comment (bacali95) [inline src/Root.tsx:163]: I suggest using the `GrInProgress` icon, it is more significant.

File: src/components/Progressbar.tsx (hunk 2 above)
Comment (k2glyph) [reply src/components/Progressbar.tsx]: Made the progress props required

File: src/Root.tsx [hunk 8]
```diff
@@ -16,6 +16,8 @@ import {
   HiStar,
   HiUser,
 } from 'react-icons/hi';
+import { AiOutlineMinus } from 'react-icons/ai';
+
```
[the remark this answers is not in the document]
Comment (k2glyph) [reply src/Root.tsx]: Removed

**Comment by k2glyph** [comment]
@bacali95 Can you please review my latest changes?

File: src/Root.tsx (hunk 7 above)
Comment (k2glyph) [reply src/Root.tsx]: done

File: src/Root.tsx (hunk 3 above)
Comment (k2glyph) [reply src/Root.tsx]: done

File: src/components/Progressbar.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  color = 'blue',
+  size = 'md',
+  label,
+  labelPosition = 'inside',
+  labelProgress = false,
+  labelSize = 'md',
+  labelColor = 'blue',
+}) => {
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (k2glyph) [reply src/components/Progressbar.tsx]: done

File: src/pages/ProgressbarPage.tsx (hunk 5 above)
Comment (k2glyph) [reply src/pages/ProgressbarPage.tsx]: done

**Review by bacali95 — APPROVED** [review]

**Comment by k2glyph** [comment]
@bacali95 I don't see my code in main branch. Why?

**Comment by bacali95** [comment]
> @bacali95 Can you please review my latest changes?

I just did some refactoring to align with the flowbite documentation, check under `components/progress`

**Pull request merged**